akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -55,6 +58,11 @@
      */
     ResponseEntity entity();
 
+    /**
+     *  Drains entity stream of this message
+     */
+    CompletionStage<Done> ignoreDataBytesJ(Materializer materializer);
```

Comment:
Regarding naming, maybe there is a way to keep the same names for both API. This is a bit tricky, I'm not sure if we want to follow this way, but seems to work. It makes use of implicit class for scala implementation instead of direct one.

```scala
implicit class HttpMessageIgnoreDataBytes(val httpMessage: HttpMessage) extends AnyVal {
  /** Drains entity stream */
  def ignoreDataBytes()(implicit m: Materializer): Future[Done] =
    httpMessage.entity.dataBytes.runWith(Sink.ignore)
}
```

Then we have:

* `_.ignoreDataBytes()` - scala version, requires implicit materializer
* `_.ignoreDataBytes(mat)` - java version, materializer passed explicitly
* `_.ignoreDataBytes()(mat)` - scala version, materializer passed explicitly

WDYT?